pkg/controller/external.go

```diff
@@ -192,7 +195,7 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 	return managed.ExternalObservation{
 		ResourceExists:          true,
 		ResourceUpToDate:        plan.UpToDate,
-		ResourceLateInitialized: lateInitialized,
+		ResourceLateInitialized: annotationsUpdated || lateInitedParams,
```

Comment:
If I'm not missing anything, we can always set this to `False` because we will already have late-inited the managed resource in the above block (maybe after a status update iteration).